System/Posix/Files/Common.hsc

```diff
@@ -294,6 +305,19 @@ modificationTime (FileStatus stat) =
 statusChangeTime (FileStatus stat) =
   unsafePerformIO $ withForeignPtr stat $ (#peek struct stat, st_ctime)
 
+#ifdef HAVE_STRUCT_STAT_ST_BLOCKS
+fileBlocks (FileStatus stat) =
+  Just $ unsafePerformIO $ withForeignPtr stat $ (#peek struct stat, st_blocks)
```

Comment:
Here, there may be a slight difference in strictness between the above, and:
```suggestion
  unsafePerformIO $ fmap Just $ withForeignPtr stat $ (#peek struct stat, st_blocks)
```
Forcing the suggested variant should (I think) actually perform the IO before returning `Just` the result.  Whereas with the original version we may still have a lazy value.  Which version do you think is the more appropriate?

Reply:
Currently GHC knows constructor in compile time, and can easily substitute it further, eliminating wrapper and associated allocations altogether. I assume that usually a result is almost immediately consumed by `fromMaybe (error "unsupported")` or similar. Swapping `Just` and `unsafePerformIO` will make constructor opaque to GHC, so I'd probably keep it as is.

Reply:
Fair enough, I guess. I have an instinctive distaste for pending lazy I/O, but this case pretty harmless, and as you say, in practice likely to be immediately evaluated by the caller.